### ISSUE TITLE
Include CrossEnropy loss in pretraining?

### ISSUE THREAD

**Issue by yaohuicai** [facebookresearch/mae#24]
According to the paper, the reconstruction loss is the only loss used in pretraining. However, I noticed that the `cls` token is appended in the input sequence of the decoder, but is dropped after the decoder.

https://github.com/facebookresearch/mae/blob/6a2ba402291005b003a70e99f7c87d1a2c376b0d/models_mae.py#L193-L194

I am wondering if you have tried to include the cross-entropy loss between the prediction of the `cls` token and the ground truth label during pretraining. Just like the NSP (next sentence prediction) task in the BERT training, intuitively, including this cross-entropy loss helps the model learns more high-level information.

**Comment by BaohaoLiao** [comment]
> According to the paper, the reconstruction loss is the only loss used in pretraining. However, I noticed that the `cls` token is appended in the input sequence of the decoder, but is dropped after the decoder.
> 
> https://github.com/facebookresearch/mae/blob/6a2ba402291005b003a70e99f7c87d1a2c376b0d/models_mae.py#L193-L194
> 
> I am wondering if you have tried to include the cross-entropy loss between the prediction of the `cls` token and the ground truth label during pretraining. Just like the NSP (next sentence prediction) task in the BERT training, intuitively, including this cross-entropy loss helps the model learns more high-level information.

But your method would become supervised learning with multi-tasks.
